pennylane/templates/state_preparations/state_prep_mps.py

```diff
@@ -15,24 +15,238 @@
+
+    n_sites = len(mps)
+    output_mps = [None] * n_sites
+
+    # Procedure analogous to the left-canonical conversion but starting from the right and storing the Vd,
+    # where Vd is the right matrix in the Singular Value Decomposition (SVD)
+    for i in range(n_sites - 1, 0, -1):
+        chi_left, d, chi_right = mps[i].shape
+        input_matrix = mps[i].reshape(chi_left, d * chi_right)
+
+        u_matrix, s_diag, vd_matrix = qml.math.linalg.svd(input_matrix, full_matrices=False)
+
+        # Truncate SVD components if needed
+        chi_new = min(int(max_bond_dim), len(s_diag))
+        u_matrix = u_matrix[:, :chi_new]
+        s_diag = s_diag[:chi_new]
+        vd_matrix = vd_matrix[:chi_new, :]
+
+        # Store Vd reshaped as an MPS tensor in the output MPS
+        output_mps[i] = vd_matrix.reshape(chi_new, d, chi_right)
+
+        # Contract U with diag(S) and merge it with the preceding MPS tensor, preserving the canonical structure
+        mps[i - 1] = qml.math.tensordot(
+            mps[i - 1], u_matrix @ qml.math.diag(s_diag), axes=([2], [0])
+        )
+
+    output_mps[0] = mps[0][0]
+    output_mps[-1] = output_mps[-1][:, :, 0]
+
+    return output_mps
+
+
 class MPSPrep(Operation):
     r"""Prepares an initial state from a matrix product state (MPS) representation.
 
     .. note::
 
-        Currently, this operator can only be used with ``qml.device(“lightning.tensor”)``.
-
+        This operator is natively supported on the ``lightning.tensor`` device, which is designed to run MPS
+        structures efficiently. For other devices, this operation prepares the state vector represented by the
+        MPS using a gate-based decomposition from Eq. (23) in `arXiv:2310.18410
+        <https://arxiv.org/pdf/2310.18410>`_, which requires the right canonicalization of the MPS using
+        the :func:`~.right_canonicalize_mps` function and defining auxiliary qubits with ``work_wires``.
 
     Args:
-        mps (List[Array]):  list of arrays of rank-3 and rank-2 tensors representing an MPS state as a
-            product of site matrices. See the usage details section for more information.
+        mps (list[TensorLike]):  list of arrays of rank-3 and rank-2 tensors representing an MPS state
+            as a product of site matrices. See the usage details section for more information.
 
         wires (Sequence[int]): wires that the template acts on
+        work_wires (Sequence[int]): list of extra qubits needed in the decomposition. If the maximum dimension
+            of the MPS tensors is :math:`2^k`, then :math:`k` ``work_wires`` will be needed. If no ``work_wires`` are given,
+            this operator can only be executed on the ``lightning.tensor`` device. Default is ``None``.
+
+        right_canonicalize (bool): indicates whether a conversion to right-canonical form should be performed to the MPS.
+            Default is ``False``.
+
+
+    .. seealso:: :func:`~.right_canonicalize_mps`.
 
     **Example**
```

Comment:
Maybe we can add an example that uses work_wires, either here or further down in the usage details.